File: Data Structures & Algorithms/duplicate-integer/submission-0.py
from typing import List

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = set()

        for num in nums:

            if num in seen:
                return True
            seen.add(num)

        return False


        



s = Solution()
result = s.hasDuplicate([1, 2, 3, 4, 1])
print(result)

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;

        for(int i=0; i<nums.size(); i++){
            int needed = target - nums[i];

            if(seen.count(needed)){
               return {seen[needed], i};
            }

            seen[nums[i]]=i;
        }
        
    }
};
